lib/spack/spack/test/cmd/list.py

```diff
@@ -64,10 +64,39 @@ def test_list_without_filters(self, parser, pkg_names):
         assert 'cloverleaf3d' in pkg_names
         assert 'hdf5' in pkg_names
 
-    def test_list_with_filters(self, parser, pkg_names):
+    def test_list_filter(self, parser, pkg_names):
+        args = parser.parse_args(['py-*'])
+        spack.cmd.list.list(parser, args)
+
+        assert pkg_names
+        assert 'py-numpy' in pkg_names
+        assert 'perl-file-copy-recursive' not in pkg_names
+
+        args = parser.parse_args(['py-'])
+        spack.cmd.list.list(parser, args)
+
+        assert pkg_names
+        assert 'py-numpy' in pkg_names
+        assert 'perl-file-copy-recursive' in pkg_names
+
+    def test_list_search_description(self, parser, pkg_names):
+        args = parser.parse_args(['--search-description', 'xml'])
+        spack.cmd.list.list(parser, args)
+
+        assert pkg_names
+        assert 'expat' in pkg_names
+
+    def test_list_tags(self, parser, pkg_names):
         args = parser.parse_args(['--tags', 'proxy-app'])
         spack.cmd.list.list(parser, args)
 
         assert pkg_names
         assert 'cloverleaf3d' in pkg_names
         assert 'hdf5' not in pkg_names
+
+    def test_list_formatter(self, parser, pkg_names):
```

Comment:
@adamjstewart IMO it would be worth adding a TODO here to test the output of the commands (that doesn't have to be done here but this could be improved beyond coverage).